dts: bindings: usb: Change the property names in the DTS

rruuaanng opened the pull request:
Change the property names in the bindings and DTS
to use hyphens(-) for separation instead of underscores(_).

kartben reviewed (changes requested):
same two comments as #83840, both for commit message being "backwards", and need for migration guide entry

rruuaanng committed:
dts: bindings: usb: Change the property names in the DTS

Change the property names in the bindings and DTS
to use hyphens(-) for separation instead of underscores(_).

Signed-off-by: James Roy <rruuaanng@outlook.com>

rruuaanng committed:
doc: Add a new migration guide entry to USB

Rename the phy_handle property to phy-handle in
the devicetree and bindings of the USB subsystem.

Signed-off-by: James Roy <rruuaanng@outlook.com>
